ai-catalog/samples/genai-image-description/src/main/java/com/android/ai/samples/genai_image_description/GenAIImageDescriptionScreen.kt

```diff
@@ -15,155 +15,222 @@
+                if (uiState is GenAIImageDescriptionUiState.Initial) {
+                    GenerateButton(
+                        text = stringResource(R.string.genai_image_description_run_inference),
+                        icon = painterResource(com.android.ai.uicomponent.R.drawable.ic_ai_edit),
+                        modifier = Modifier
+                            .align(Alignment.BottomStart)
+                            .padding(start = 24.dp, bottom = 24.dp),
+                    ) {
+                        onGenerateClick(imageUri)
+                    }
+                }
+            } else {
+                PrimaryButton(
+                    text = stringResource(R.string.genai_image_description_add_image),
+                    icon = painterResource(id = com.android.ai.uicomponent.R.drawable.ic_ai_img),
+                    modifier = Modifier
+                        .height(96.dp)
+                        .padding(start = 24.dp, end = 24.dp)
+                        .align(Alignment.Center),
+                    onClick = onImagePickerClick,
                 )
             }
 
-            // Generate image description button
-            Button(
-                onClick = {
-                    showBottomSheet = true
-                    viewModel.getImageDescription(imageUri, context)
-                }, modifier = Modifier.padding(10.dp).align(Alignment.CenterHorizontally),
+            if (
+                uiState !is GenAIImageDescriptionUiState.Initial
+                && uiState !is GenAIImageDescriptionUiState.CheckingFeatureStatus
             ) {
-                Text(
-                    text = stringResource(id = R.string.genai_image_description_run_inference),
-                )
-            }
-        }
+                val outputText = when (val state = uiState) {
+                    is GenAIImageDescriptionUiState.DownloadingFeature -> stringResource(
+                        id = R.string.image_desc_downloading,
+                        state.bytesDownloaded,
+                        state.bytesToDownload,
+                    )
+                    is GenAIImageDescriptionUiState.Error -> stringResource(state.errorMessageStringRes)
+                    is GenAIImageDescriptionUiState.Generating -> state.partialOutput
+                    is GenAIImageDescriptionUiState.Success -> state.generatedOutput
+                    else -> "" // Show nothing for the Initial state
+                }
 
-        if (showBottomSheet) {
-            ModalBottomSheet(
-                onDismissRequest = {
-                    showBottomSheet = false
-                    viewModel.clearGeneratedText()
-                }, sheetState = sheetState,
-            ) {
-                Text(
-                    text = imageDescriptionResult.value,
-                    modifier = Modifier.padding(top = 8.dp, bottom = 24.dp, start = 24.dp, end = 24.dp),
+                SecondaryButton(
```

Comment:
shouldn't be addressed in this PR, but I don't think there's currently a way to add content descriptions for buttons that only have icons, which makes it unusable for users of accessibility services.